Change datatypes

trevorcummingsCCS opened the pull request:
Changed lot_organisation_roles, lot_peole_roles, commercial_agreement_organisational_roles,contact_details and contact_points data type columns where we were using TIMESTAMP to DATE. Plus changed Business Type in Organisations from integer to string.

trevorcummingsCCS committed:
Change datatypes

trevorcummingsCCS committed:
Update drop_tables.sql

Added extra drop table command for new table added

trevorcummingsCCS committed:
Added additional contact point tables

trevorcummingsCCS committed:
Not all table names changed on index creation statements

trevorcummingsCCS committed:
Table name invalid

Table contact_points_lot_ors should have been contact_point_lot_ors

trevorcummingsCCS committed:
syntax error

trevorcummingsCCS committed:
miss spelt column name

trevorcummingsCCS committed:
Added drop commands for new tables

trevorcummingsCCS committed:
Create create_commercial_agreement_documents.sql

trevorcummingsCCS committed:
added additional scripts

create_commercial_agreement_documents
create_contact_point_lot_prs
create_contact_point_lot_ors
create_contact_point_commercial_agreement_ors

trevorcummingsCCS committed:
Create create_contact_point_lot_prs.sql

trevorcummingsCCS committed:
Create create_contact_point_lot_ors.sql

trevorcummingsCCS committed:
Create create_contact_point_commercial_agreement_ors.sql

tommyb82 commented on agreements/provision-agreements-database.sh:
```diff
@@ -25,7 +25,9 @@ psql -h $SERVER -d $DATABASE -p $PORT -U $USERNAME -a -q -f create_commercial_ag
 psql -h $SERVER -d $DATABASE -p $PORT -U $USERNAME -a -q -f create_contact_details.sql
 psql -h $SERVER -d $DATABASE -p $PORT -U $USERNAME -a -q -f create_contact_point_reasons.sql
-psql -h $SERVER -d $DATABASE -p $PORT -U $USERNAME -a -q -f create_contact_points.sql
+psql -h $SERVER -d $DATABASE -p $PORT -U $USERNAME -a -q -f create_contact_point_lot_prs.sql
```
Ordering error:
```
psql:create_contact_point_lot_prs.sql:8: ERROR:  insert or update on table "contact_point_lot_prs" violates foreign key constraint "contact_point_lot_prs_lot_people_roles_fk"
DETAIL:  Key (lot_person_role_id)=(1) is not present in table "lot_people_roles".
```

trevorcummingsCCS committed:
Moved contact point reason scripts

tommyb82 reviewed (approved):
Grand! 👍

trevorcummingsCCS committed:
Role Types and Bigints

Consolidated three type tables into one role type table. Created an ENUM and populated it. Change reference to BIGINT and BIGSERIAL to INTEGER and SERIAL

trevorcummingsCCS committed:
Consilidation of

trevorcummingsCCS committed:
consolidated role types

trevorcummingsCCS committed:
Update create_commercial_agreement_role_types.sql

trevorcummingsCCS committed:
Create create_role_types.sql

trevorcummingsCCS committed:
Delete create_commercial_agreement_role_types.sql

Consolidated role type tables

trevorcummingsCCS committed:
Delete create_lot_people_role_types.sql

Consolidated role type tables

trevorcummingsCCS committed:
Delete create_organisation_role_types.sql

Consolidated role type tables

trevorcummingsCCS committed:
Update create_commercial_agreement_organisation_roles.sql

Consolidated role type tables

trevorcummingsCCS committed:
Update create_lot_organisation_roles.sql

Consolidated role type tables

trevorcummingsCCS committed:
Added entity_id to Organisations

trevorcummingsCCS committed:
Update create_organisations.sql

tommyb82 commented:
Hi @trevorcummingsCCS - thanks for the recent update to include organisation `entity_id` 👍 
Couple more observations:

- Organisation `scheme` does not appear in the DB but is required by OCDS (an enum of `[ GB-COH, GB-MPR, GB-NIC, GB-CHC, GB-SC, GB-WALEDU, GB-SCOTEDU, GB-GOR, GB-LANI, GB-NHS ]`)
- Could we include the `status` ("active", "suspended" or "excluded") in the organisations data insert

Might be a few more tomorrow... 😎

trevorcummingsCCS committed:
Update ddl.sql

Denormalised contact method type on contact details. This means that contact details holds a full set of contact information. Added contact name to the contact point tables

trevorcummingsCCS committed:
Update create_contact_details.sql

trevorcummingsCCS committed:
Update create_contact_method_types.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update create_contact_details.sql

trevorcummingsCCS committed:
Delete create_contact_method_types.sql

trevorcummingsCCS committed:
Update drop_tables.sql

trevorcummingsCCS committed:
Update provision-agreements-database.sh
